### ISSUE TITLE
`ps -f before=`  vs  `ps -f since=` 

### ISSUE THREAD

**Issue by ozlerhakan** [docker/docker#20087]
Hi all,

`$docker ps -f since=` shows a result with and without `-a`, Next to this, after trying `$docker ps -f before=box`, I got nothing, but this query works with `-a`; `$docker ps -af before=box`. So while I use `since`  with two options, why I cannot use `before` without `-a`? or  does `since` have bad behavior by providing both cases?  

How reproducible:
```
$ docker ps -a
CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS                      PORTS               NAMES
aa99c1c1e96b        ubuntu:14.04              "bash"                   3 hours ago         Exited (0) 3 hours ago                          jolly_ritchie
8c6bd18f842e        busybox:latest            "sleep 100"              4 hours ago         Exited (143) 4 hours ago                        box
168cbe37008c        mongo:3.2                 "/entrypoint.sh mongo"   2 days ago          Exited (137) 30 hours ago   27017/tcp           mongo
1c60d589fcf6        centurylink/image-graph   "/usr/src/app/image-g"   4 days ago          Exited (137) 4 days ago                         happy_leavitt
b76fb791a1df        dockerui/dockerui         "/dockerui"              5 days ago          Exited (2) 4 days ago                           compassionate_bell

$ docker ps -f since=box
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS                   PORTS               NAMES
aa99c1c1e96b        ubuntu:14.04        "bash"              3 hours ago         Exited (0) 3 hours ago                       jolly_ritchie

$ docker ps -af since=box
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS                   PORTS               NAMES
aa99c1c1e96b        ubuntu:14.04        "bash"              3 hours ago         Exited (0) 3 hours ago                       jolly_ritchie

$ docker ps -f before=box
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES

$ docker ps -af before=box
CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS                      PORTS               NAMES
168cbe37008c        mongo:3.2                 "/entrypoint.sh mongo"   2 days ago          Exited (137) 30 hours ago   27017/tcp           mongo
1c60d589fcf6        centurylink/image-graph   "/usr/src/app/image-g"   4 days ago          Exited (137) 4 days ago                         happy_leavitt
b76fb791a1df        dockerui/dockerui         "/dockerui"              5 days ago          Exited (2) 4 days ago                           compassionate_bell

```

`docker version`:
```
Client:
 Version:      1.10.0
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   590d5108
 Built:        Thu Feb  4 18:18:11 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.10.0
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   590d5108
 Built:        Thu Feb  4 19:55:25 2016
 OS/Arch:      linux/amd64
```

`docker info`:
```
Containers: 5
 Running: 0
 Paused: 0
 Stopped: 5
Images: 103
Server Version: 1.10.0
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 122
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: host bridge null
Kernel Version: 4.1.17-boot2docker
Operating System: Boot2Docker 1.10.0 (TCL 6.4.1); master : b09ed60 - Thu Feb  4 20:16:08 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 996.2 MiB
Name: kodcu
ID: 32KK:K7BX:D2KV:SXYC:N65U:2FBX:IUQD:U24R:NBYR:EEV4:BBPU:Y66D
Debug mode (server): true
 File Descriptors: 10
 Goroutines: 22
 System Time: 2016-02-07T14:20:50.732065775Z
 EventsListeners: 0
 Init SHA1:
 Init Path: /usr/local/bin/docker
 Docker Root Dir: /mnt/sda1/var/lib/docker
Username: 
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
```

**Comment by thaJeztah** [comment]
Thanks for reporting; since all containers are exited, I think the `since` filter is the wrong behavior here; filter should only *filter* the results, not add extra items to the result.

**Comment by thaJeztah** [comment]
/cc @wenchma 

I see this was implemented in https://github.com/docker/docker/pull/17718; I suspect the separate `--since` flag acts like this (not checked yet), but for a filter it should act like a filter.

Added to the 1.10.1 milestone (but may slip to the next one), because this was added in 1.10

**Comment by vdemeester** [comment]
@thaJeztah yeah, I think this is more or less how `--since` and `-before` were working (looking at the changes). I'll take a look :stuck_out_tongue_closed_eyes:

**Comment by calavera** [comment]
Unfortunately, this is the expected behavior, copying what `--before` does. I agree that it's inconsistent and we should fix it. But I removed from the milestone because it's not a regression.

**Comment by thaJeztah** [comment]
@calavera reason I added, is that it's a bug in a new feature (it's new as filter in 1.10), so I'd prefer to have it fixed in a 1.10.x patch release to prevent people starting to rely on the incorrect behavior

**Comment by vdemeester** [comment]
@calavera @thaJeztah I'm fine with both, I have a fix, a PR is coming soon (tonight or tomorrow morning, europe time :angel:)
